src/config.js

```diff
@@ -12,20 +12,46 @@ config.save = function () {
 };
 
 config.addUser = function(GUILD_ID, ROLE_ID, USER_ID) {
-    config.data[GUILD_ID][ROLE_ID].members.push(USER_ID)
-    config.save();
+  return new Promise((resolve, reject) =>{
+    try{
+      for(let i = 0; i < config.data[GUILD_ID].games.length; i++){
+        if(config.data[GUILD_ID][ROLE_ID].game == config.data[GUILD_ID].games[i][0]){
+          if(config.data[GUILD_ID][ROLE_ID].members.length < config.data[GUILD_ID].games[i][1]){
+            if(config.data[GUILD_ID][ROLE_ID].members.includes(USER_ID) == false){ //Makes sure there will be no duplicate entries in the player list
+              config.data[GUILD_ID][ROLE_ID].members.push(USER_ID)
+              config.save()
+              resolve(true)
+            }
+          }else if(config.data[GUILD_ID][ROLE_ID].members.length == config.data[GUILD_ID].games[i][1]){
+            //MAYBE PROACTIVELY SEND A MESSAGE TO THE GUILD IF THE GROUP IS FULL?
+          }else{
+            //GROUP IS FULL. DO WHATEVER
+            console.log('Group full')
+            reject('full')
+          }
+        }
+      }
+    }catch(err){
+
+    }
+  })
+
+
 };
 
-/*config.removeUser = function (guild, user) {
-    delete guild.users[user.id];
-    config.save();
-}; Needs to be re-worked */
+config.removeUser = function(GUILD_ID, ROLE_ID, USER_ID) {
+  if(USER_ID != config.data[GUILD_ID][ROLE_ID].creator){
+    delete config.data[GUILD_ID][ROLE_ID].members.splice(config.data[GUILD_ID][ROLE_ID].members.indexOf(USER_ID), 1)
+    config.save()
+  }
+};
 
 config.createSession = function(GUILD_ID,USER_ID, ROLE_ID, GAME, CHANNEL_ID, MESSAGE_ID) {
     if (config.data[GUILD_ID] === undefined) {
         config.data[GUILD_ID] = {}
     }
     config.data[GUILD_ID][ROLE_ID] = {
+        creator: USER_ID,
```

Comment:
I'm not sure if we should treat the creator any differently than the other users. Nobody can be added twice and I think the creator should be allowed to leave their room, if they want to cancel it or something.